sahi/models/base.py

```diff
@@ -57,7 +60,8 @@ def __init__(
         self._original_predictions = None
         self._object_prediction_list_per_image = None
 
-        self.set_device()
+        if not ".xml" == self.model_path[-4:]:
+            self.set_device()
```

Comment:
Please don't change the base class, move these operations to Yolov8OpenvinoDetectionModel class